navigation/navigation-safe-args-generator/src/main/kotlin/androidx/navigation/safe/args/generator/java/JavaNavWriter.kt

```diff
@@ -248,6 +248,52 @@ class JavaNavWriter(private val useAndroidX: Boolean = true) : NavWriter<JavaCod
             addStatement("return $N", result)
         }.build()
 
+        val fromSavedStateHandleMethod = MethodSpec.methodBuilder("fromSavedStateHandle").apply {
+            addAnnotation(annotations.NONNULL_CLASSNAME)
+            addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+            addAnnotation(specs.suppressAnnotationSpec)
+            val savedStateHandle = "savedStateHandle"
+            addParameter(
+                ParameterSpec.builder(SAVED_STATE_HANDLE_CLASSNAME, savedStateHandle)
+                    .addAnnotation(specs.androidAnnotations.NONNULL_CLASSNAME)
+                    .build()
+            )
+            returns(className)
+            val result = "__result"
+            addStatement("$T $N = new $T()", className, result, className)
+            args.forEach { arg ->
+                beginControlFlow("if ($N.contains($S))", savedStateHandle, arg.name).apply {
+                    addStatement("$T $N", arg.type.typeName(), arg.sanitizedName)
```

Comment:
Looks like the variable declaration and initialization can be done in the same statement

Reply:
Not anymore with the newest changes using the common `addReadSingleArgBlock` function 😄